### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -7,38 +7,40 @@
...
 provider/core/                    @sffc @Manishearth
 provider/datagen/                 @sffc @robertbastian @Manishearth
 provider/fs/                      @sffc
 provider/macros/                  @Manishearth @sffc
-provider/testdata/                @unicode-org/icu4x-owners
 tools/benchmark/binsize/          @gnrunge
 tools/benchmark/macros/           @gregtatum
 tools/benchmark/memory/           @gregtatum
 tools/scripts/                    @Manishearth @sffc
+utils/databake                    @robertbastian
```
Comment (Manishearth) [inline CODEOWNERS:37]: +me please

File: CODEOWNERS [hunk 2]
```diff
@@ -7,38 +7,40 @@
...
 experimental/bies/                @sffc
-experimental/calendar/            @Manishearth @sffc
-experimental/codepointtrie/       @echeran
-experimental/provider_ppucd/      @echeran
+experimental/compactdecimal/      @eggrobin
+experimental/displaynames/        @sffc
+experimental/relativetime/        @pdogr
 experimental/segmenter/           @aethanyc @makotokato @sffc
 ffi/capi/                         @Manishearth
-ffi/cpp/                          @Manishearth
```
Comment (Manishearth) [inline CODEOWNERS:26]: should probably add a blanket ffi/ instead

File: CODEOWNERS [hunk 3]
```diff
@@ -7,38 +7,40 @@
...
 components/plurals/               @zbraniecki @sffc
-components/uniset/                @echeran @iainireland
+components/timezone/              @nordzilla
 experimental/bies/                @sffc
-experimental/calendar/            @Manishearth @sffc
-experimental/codepointtrie/       @echeran
-experimental/provider_ppucd/      @echeran
+experimental/compactdecimal/      @eggrobin
+experimental/displaynames/        @sffc
+experimental/relativetime/        @pdogr
```
Comment (Manishearth) [inline CODEOWNERS:24]: potentially add someone else here if there is someone that makes sense (if possible I'd like there to be two component owners especially for situations where one owner is working on icu4x on the side)

Comment (robertbastian) [reply CODEOWNERS:24]: Who? One owner is better than nobody.

File: CODEOWNERS (hunk 2 above)
Comment (robertbastian) [reply CODEOWNERS:26]: Everything that's not listed defaults to `icu4x-owners`, so I think it's fine?

Comment (Manishearth) [reply CODEOWNERS:26]: Mostly to say who the explicit point of contact is

File: CODEOWNERS (hunk 3 above)
Comment (Manishearth) [reply CODEOWNERS:24]: Oh, absolutely, but as long as we're adding it we could potentially add two.

Then again it seems like it's mostly been us reviewing Pawan's work so I'm fine leaving it as just him for now

File: CODEOWNERS [hunk 4]
```diff
@@ -7,38 +7,40 @@
 
 # List of components with at least two owners as specified in docs/crate-ownership.
 
+components/calendar/              @Manishearth @sffc
 components/collator/              @hsivonen @echeran
+components/collections/           @echeran @iainireland
```
Comment (sffc) [inline CODEOWNERS:12]: @iainireland are you happy to be a code owner here?

File: CODEOWNERS [hunk 5]
```diff
@@ -7,38 +7,40 @@
...
 components/normalizer/            @hsivonen @echeran
 components/plurals/               @zbraniecki @sffc
-components/uniset/                @echeran @iainireland
+components/timezone/              @nordzilla
 experimental/bies/                @sffc
-experimental/calendar/            @Manishearth @sffc
-experimental/codepointtrie/       @echeran
-experimental/provider_ppucd/      @echeran
+experimental/compactdecimal/      @eggrobin
+experimental/displaynames/        @sffc
```
Comment (sffc) [inline CODEOWNERS:23]: Please add Sanket

File: CODEOWNERS [hunk 6]
```diff
@@ -7,38 +7,40 @@
...
 tools/benchmark/macros/           @gregtatum
 tools/benchmark/memory/           @gregtatum
 tools/scripts/                    @Manishearth @sffc
+utils/databake                    @robertbastian
+utils/deduplicating_array         @robertbastian
 utils/fixed_decimal/              @sffc
 utils/litemap/                    @Manishearth @sffc
 utils/pattern/                    @zbraniecki
-utils/writeable/                  @sffc
+utils/tinystr                     @zbraniecki
```
Comment (sffc) [inline CODEOWNERS:42]: +me

**Review by sffc — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**